js/index.js

```diff
@@ -2,128 +2,798 @@
 
 (function () {
   'use strict';
-
+  global.jQuery = require("jquery")
   var avsc = require('avsc'),
       buffer = require('buffer'),
+      utils = require('./utils'),
       $ = require('jquery');
+  window.avsc = avsc;
+  $( function() {
 
+    var outputErrorElement = $('#encoded-error'),
+        inputErrorElement = $('#decoded-error'),
+        encodedValidElement = $('#output-valid'),
+        decodedValidElement = $('#input-valid'),
+        schemaErrorElement = $('#schema-error'),
+        schemaValidElement = $('#schema-valid'),
+        schemaSelect = $('#schema-template'),
+        schemaElement = document.getElementById('schema'),
+        inputElement = $('#input'),
+        outputElement = $('#output'),
+        body = document.getElementsByTagName('body')[0],
+        arrayKeyPattern = /(\d+)/g,
+        queryPattern = /[?&#]+([\w]+)=([^&#]*)/g,
+        reservedKeysPattern = /-[a-z]+-/g,
+        whitespacePattern = /[\s]+/g,
+        typingTimer,
+        eventObj = utils.eventObj,
+        doneTypingInterval = 500; // wait for some time before processing user input.
+    
+    window.reverseIndexMap = [];  
 
-  require('jquery-ui');
-  window.avsc = avsc;
+    eventObj.on('schema-changed', function() {
+      runPreservingCursorPosition( 'schema', validateSchema);
+    }).on('input-changed', function() {
```

Comment:
It might make the code a bit simpler to understand if we passed arguments to the handlers (instead of relying on globals). For example here `'input-changed'` could get the trimmed input.